File: CW/Lista_1/Z1.scala
def flatten1[A](xss: List[List[A]]): List[A] = {
    if(xss.isEmpty) then
        List()
    else
        xss.head ::: flatten1(xss.tail)
}

@main def main() = {
    val listOfLists = List(List(5, 6), List(1, 2, 3))
    println(flatten1(listOfLists))
}

File: CW/Lista_1/Z1.ml
let rec flatten1 (xss: 'a list list) =
  if xss = [] then
    []
  else
    (List.hd xss) @ (flatten1 (List.tl xss))

let lst = [[5;6];[1;2;3]];;
assert (flatten1 lst = [5;6;1;2;3]);;